libraries/node-core-library/src/Executable.ts

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as child_process from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+
+import { FileSystem, PosixModeBits } from './FileSystem';
+
+/**
+ * Typings for one of the streams inside IExecutableSpawnSyncOptions.stdio.
+ * @beta
+ */
+export type ExecutableStdioStreamMapping = 'pipe' | 'ignore' | 'inherit'
+  | NodeJS.WritableStream | NodeJS.ReadableStream
+  | number | undefined;
+
+/**
+ * Typings for IExecutableSpawnSyncOptions.stdio.
+ * @beta
+ */
+export type ExecutableStdioMapping = 'pipe' | 'ignore' | 'inherit' | ExecutableStdioStreamMapping[];
+
+/**
+ * Options for Executable.tryResolve().
+ * @beta
+ */
+export interface IExecutableResolveOptions {
+  /**
+   * The current working directory.  If omitted, process.cwd() will be used.
+   */
+  currentWorkingDirectory?: string;
+
+  /**
+   * The environment variables for the child process.  If omitted, process.env will be used.
+   */
+  environment?: NodeJS.ProcessEnv;
+}
+
+/**
+ * Options for Executable.execute().
+ * @beta
+ */
+export interface IExecutableSpawnSyncOptions extends IExecutableResolveOptions {
+  /**
+   * The content to be passed to the child process's stdin.
+   */
+  input?: string;
+
+  /**
+   * The stdio mappings for the child process.
+   */
+  stdio?: ExecutableStdioMapping;
```

Comment:
What happens if we specify both an input and stdin?

Reply:
In the Node API, this takes prescience.

---
In reply to: [208417043](https://github.com/Microsoft/web-build-tools/pull/764#discussion_r208417043) [](ancestors = 208417043)

Reply:
Updated documentation

---
In reply to: [208417232](https://github.com/Microsoft/web-build-tools/pull/764#discussion_r208417232) [](ancestors = 208417232,208417043)